qmake: bindings_target race condition leads to compile error (cutterbindings_python.h)

**Environment information**
* Operating System: Arch Linux
* Cutter version: 1.10.2
* Obtained from:
  - [x] Built from source

**Describe the bug**

QMake build system doesn't seem to properly specify target dependencies. When i build on a very beefy machine like a ryzen with 64 concurrency, the compilation aborts (in like ~3 out of 4 cases) with:
```
...
g++ -c -pipe -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -std=gnu++11 -Wall -Wextra -D_REENTRANT -fPIC -DCUTTER_ENABLE_PYTHON -DCUTTER_ENABLE_PYTHON_BINDINGS -D_CRT_NONSTDC_NO_DEPRECATE -D_CRT_SECURE_NO_WARNINGS -DQT_NO_DEBUG -DQT_SVG_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I../src -I. -I../src -I../src/core -I../src/widgets -I../src/dialogs -I../src/common -I../src/plugins -I../src/menus -I/build/radare2-cutter/src/radare2-cutter/build/bindings/CutterBindings -isystem /usr/include/PySide2 -isystem /usr/include/PySide2/QtCore -isystem /usr/include/PySide2/QtWidgets -isystem /usr/include/PySide2/QtGui -isystem /usr/include/libr -isystem /usr/include/python3.8 -isystem /usr/include/shiboken2 -isystem /usr/include/qt -isystem /usr/include/qt/QtSvg -isystem /usr/include/qt/QtWidgets -isystem /usr/include/qt/QtGui -isystem /usr/include/qt/QtNetwork -isystem /usr/include/qt/QtCore -I. -I. -I/usr/lib/qt/mkspecs/linux-g++ -o BasicBlockHighlighter.o ../src/common/BasicBlockHighlighter.cpp
../src/plugins/PluginManager.cpp:6:10: fatal error: cutterbindings_python.h: No such file or directory
    6 | #include <cutterbindings_python.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:5239: PluginManager.o] Error 1
make: *** Waiting for unfinished jobs....
```

I can work around this issue by either limiting the parallel jobs (hence making it more unlikely to out-race the binding generator) or by simply invoking `make bindings_target` myself before the full `make` call.

**To Reproduce**

```
  qmake ../src/Cutter.pro \
    CUTTER_ENABLE_PYTHON=true \
    CUTTER_ENABLE_PYTHON_BINDINGS=true \
    CUTTER_USE_BUNDLED_RADARE2=false
  make
```

**Expected behavior**

Can compile using qmake on a beefy machine without race conditions.

--- a/src/Cutter.pro
+++ b/src/Cutter.pro
@@ -156,6 +156,7 @@ CUTTER_ENABLE_PYTHON {
         }
         BINDINGS_SRC_DIR = "$${PWD}/bindings"
         BINDINGS_BUILD_DIR = "$${OUT_PWD}/bindings"
+        BINDINGS_SOURCE_DIR = "$${BINDINGS_BUILD_DIR}/CutterBindings"
         BINDINGS_SOURCE = $$system("$${BINDINGS_SRC_LIST_CMD} qmake \"$${BINDINGS_BUILD_DIR}\"")
         BINDINGS_INCLUDE_DIRS = "$$[QT_INSTALL_HEADERS]" \
                                 "$$[QT_INSTALL_HEADERS]/QtCore" \
@@ -194,9 +195,9 @@ CUTTER_ENABLE_PYTHON {
         bindings.commands = $$quote($$system_path($${SHIBOKEN_EXECUTABLE})) $${SHIBOKEN_OPTIONS}
         QMAKE_EXTRA_TARGETS += bindings
         PRE_TARGETDEPS += bindings_target
-        GENERATED_SOURCES += $${BINDINGS_SOURCE}
+        # GENERATED_SOURCES += $${BINDINGS_SOURCE} done by dummy targets bellow
 
-        INCLUDEPATH += "$${BINDINGS_BUILD_DIR}/CutterBindings"
+        INCLUDEPATH += "$${BINDINGS_SOURCE_DIR}"
 
         win32:DEFINES += WIN32_LEAN_AND_MEAN
 
@@ -211,6 +212,34 @@ CUTTER_ENABLE_PYTHON {
             PKGCONFIG += shiboken2 pyside2
         }
         INCLUDEPATH += "$$PYSIDE_INCLUDEDIR" "$$PYSIDE_INCLUDEDIR/QtCore" "$$PYSIDE_INCLUDEDIR/QtWidgets" "$$PYSIDE_INCLUDEDIR/QtGui"
+
+
+        BINDINGS_DUMMY_INPUT_LIST = bindings/src_list.py
+
+        # dummy rules to specify dependency between generated binding files and bindings_target
+        bindings_h.input = BINDINGS_DUMMY_INPUT_LIST
+        bindings_h.depends = bindings_target
+        bindings_h.output = cutterbindings_python.h
+        bindings_h.commands = "echo placeholder command ${QMAKE_FILE_OUT}"
+        bindings_h.variable_out = HEADERS
+        QMAKE_EXTRA_COMPILERS += bindings_h
+
+        for(path, BINDINGS_SOURCE) {
+            dummy_input = $$replace(path, .cpp, .txt)
+            BINDINGS_DUMMY_INPUTS += $$dummy_input
+            win32 {
+                _ = $$system("mkdir \"$$dirname(dummy_input)\"; echo a >\"$$dummy_input\"")
+            } else {
+                _ = $$system("mkdir -p \"$$dirname(dummy_input)\"; echo a >\"$$dummy_input\"")
+            }
+        }
+
+        bindings_cpp.input = BINDINGS_DUMMY_INPUTS
+        bindings_cpp.depends = bindings_target
+        bindings_cpp.output = "$${BINDINGS_SOURCE_DIR}/${QMAKE_FILE_IN_BASE}.cpp"
+        bindings_cpp.commands = "echo placeholder command ${QMAKE_FILE_OUT}"
+        bindings_cpp.variable_out = GENERATED_SOURCES
+        QMAKE_EXTRA_COMPILERS += bindings_cpp
     }
 }
 

